src/main/java/io/split/android/client/service/sseclient/sseclient/BackoffCounterTimer.java

```diff
@@ -33,15 +34,18 @@ public void cancel() {
         }
         mStreamingBackoffCounter.resetCounter();
         mTaskExecutor.stopTask(mTaskId);
+        mTaskId = null;
     }
 
     public void schedule() {
-        if(mTask == null) {
+        // mTaskId != null means task already scheduled, so return to avoid schedule a second one
+        if(mTask == null || mTaskId != null) {
             return;
         }
 
-        cancel();
-        mTaskId = mTaskExecutor.schedule(mTask, mStreamingBackoffCounter.getNextRetryTime(), this);
+        long retryTime = mStreamingBackoffCounter.getNextRetryTime();
+        Logger.d(String.format("Retrying streaming reconnection in %d seconds", retryTime));
```

Comment:
this log is ok, except the BackoffCounterTimer is used as a util somewhere else.

Reply:
Yup doc, updated!